src/client/dfs/dfs.c

```diff
@@ -3514,6 +3526,20 @@ dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 	return rc;
 }
 
+int
+dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
+	    struct dirent *dirs)
+{
+	return readdir_int(dfs, obj, anchor, nr, dirs, NULL);
+}
+
+int
+dfs_readdirplus(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
+		struct dirent *dirs, struct stat *stbufs)
+{
+	return readdir_int(dfs, obj, anchor, nr, dirs, stbufs);
+}
```

Comment:
Do we need a NULL check on `stbufs`?

Reply:
i made it as optionally populated if not NULL:
https://github.com/daos-stack/daos/pull/10438/files#diff-dd80cb8b1b118e01084005bad5c40fabe2a4b05744db070f7f3dc2142aa98c60R3505

Reply:
Right. I guess if it's NULL, it's the same as calling `dfs_readdir`